committeeoversightapp/management/commands/import_data.py

```diff
@@ -115,98 +109,165 @@ def add_house_hearings(self):
 
-                    row_count += 1
+                    elif name:
+                        self.smart_match(matched, name, committees_filtered, start_date, category, source, classification)
 
-    def get_committee(self, committee_name, parent):
+    def get_committee(self, committee_name, parent, committee_key):
         classification = "committee"
-        organization = Organization.objects.filter(Q(name__icontains=committee_name, parent=parent, classification=classification) |
-                    Q(other_names__name__icontains=committee_name, parent=parent, classification=classification))
-        if organization.count() == 1:
-            return organization[0]
-        elif organization.count() == 0:
-            return IndexError
+        organization_exact = Organization.objects.filter(name=committee_name, parent=parent, classification=classification)
+        organization_contains = Organization.objects.filter(name__icontains=committee_name, parent=parent, classification=classification)
+
+        if organization_exact.count() == 1:
+            return organization_exact[0]
         else:
-            return ValueError
+            if organization_contains.count() == 1:
+                return organization_contains[0]
+            elif organization_contains.count() == 0:
+                organization_other_name = Organization.objects.filter(other_names__name__icontains=committee_name, parent=parent, classification=classification)
+                try:
+                    return organization_other_name[0]
+                except IndexError:
+                    new_committee, _ = Committee.objects.get_or_create(lugar_id=committee_key, lugar_name=committee_name)
+                    self.bad_rows.append("Not recognized: " + committee_key + ", " + committee_name)
+            else:
+                self.bad_rows.append("Multiple possible committees for " + committee_key + ": " + committee_name)
 
-    def new_committee(self, committee_key, event, row_count):
+    def new_event_participant(self, committee_key, event):
         try:
             name = Committee.objects.get(lugar_id=committee_key).organization.name
             organization = Committee.objects.get(lugar_id=committee_key).organization
             entity_type = "organization"
             committee, created = EventParticipant.objects.get_or_create(name=name, event=event, organization=organization, entity_type=entity_type)
 
         except (ObjectDoesNotExist, AttributeError, ValueError) as e:
-            self.bad_rows.append("Row " + str(row_count) + ": Bad committee key " + committee_key)
+            self.bad_rows.append("Row " + str(self.row_count) + ": Bad committee key " + committee_key)
+
+    def new_category(self, event, category):
+        try:
+            hearing_category, _ = HearingCategory.objects.get_or_create(event=event, category_id=category)
+
+        except IntegrityError:
```

Comment:
How could there be an integrity error here?

Reply:
Some of the categories in Lugar's spreadsheet are not one of the set categories Jamie sent us  (see [fixture](https://github.com/datamade/committee-oversight/blob/master/committeeoversightapp/fixtures/hearingcategorytype.json)). 0 and 14 both show up a couple times.